proxy/src/opentsdb/types.rs

```diff
@@ -0,0 +1,226 @@
+            Value::F64Value(v) => Display::fmt(v, f),
+            Value::StringValue(v) => Display::fmt(v, f),
+        }
+    }
+}
+
+pub(crate) fn convert_put_request(req: PutRequest) -> Result<Vec<WriteTableRequest>> {
+    let points = {
+        // multi points represent as json array
+        let parse_array = from_slice::<Vec<Point>>(&req.points);
+        match parse_array {
+            Ok(points) => Ok(points),
+            Err(_e) => {
+                // single point represent as json object
+                let parse_object = from_slice::<Point>(&req.points);
+                match parse_object {
+                    Ok(point) => Ok(vec![point]),
+                    Err(e) => Err(e),
+                }
+            }
+        }
+    };
+    let points = points.box_err().with_context(|| ErrWithCause {
+        code: StatusCode::BAD_REQUEST,
+        msg: "Json parse error".to_string(),
+    })?;
+    validate(&points)?;
+
+    let mut points_per_metric = HashMap::new();
+    for point in points.into_iter() {
+        points_per_metric
+            .entry(point.metric.clone())
+            .or_insert(Vec::new())
+            .push(point);
+    }
+
+    let mut requests = Vec::with_capacity(points_per_metric.len());
+    for (metric, points) in points_per_metric {
+        let mut tag_names_set = HashSet::new();
+        for point in points.iter() {
+            for tag_name in point.tags.keys() {
+                tag_names_set.insert(tag_name.clone());
+            }
+        }
+
+        let mut tag_name_to_tag_index: HashMap<String, u32> = HashMap::new();
+        let mut tag_names = Vec::with_capacity(tag_names_set.len());
+        for (idx, tag_name) in tag_names_set.into_iter().enumerate() {
+            tag_name_to_tag_index.insert(tag_name.clone(), idx as u32);
+            tag_names.push(tag_name);
+        }
+
+        let mut req = WriteTableRequest {
+            table: metric,
+            tag_names,
+            field_names: vec![String::from(OPENTSDB_DEFAULT_FIELD)],
+            entries: Vec::with_capacity(points.len()),
+        };
+
+        for point in points {
+            let timestamp = point.timestamp;
+            let timestamp = try_to_millis(timestamp)
+                .with_context(|| ErrNoCause {
+                    code: StatusCode::BAD_REQUEST,
+                    msg: format!("Invalid timestamp: {}", point.timestamp),
+                })?
+                .as_i64();
+
+            let mut tags = Vec::with_capacity(point.tags.len());
+            for (tag_name, tag_value) in point.tags {
+                let &tag_index = tag_name_to_tag_index.get(&tag_name).unwrap();
+                tags.push(Tag {
+                    name_index: tag_index,
+                    value: Some(ProtoValue {
+                        value: Some(value::Value::StringValue(tag_value)),
+                    }),
+                });
+            }
+
+            let f64_value = point
```

Comment:
Why convert all fields to f64?

Reply:
Because OpenTSDB only supports floating point values

Reply:
http://opentsdb.net/docs/build/html/user_guide/writing/#floating-point-values

Reply:
> if a field is detect to be a integer datatype, it can't accept any floating point value anymore.

I think this is what's expected, If a value doesn't contains a `.`, it's regarded as i64.

Reply:
Now use i64 if the value is a integer